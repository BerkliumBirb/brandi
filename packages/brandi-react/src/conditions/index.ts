export * from './ConditionsContext';
export * from './ConditionsProvider';
export * from './tagged';
export * from './useConditions';
