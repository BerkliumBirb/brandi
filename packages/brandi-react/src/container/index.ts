export * from './ContainerContext';
export * from './ContainerProvider';
export * from './useContainer';
