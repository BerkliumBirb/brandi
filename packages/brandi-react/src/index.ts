export { ContainerProvider } from './container';
export { createInjectionHooks, useInjection } from './injection';
export { tagged } from './tagged';
