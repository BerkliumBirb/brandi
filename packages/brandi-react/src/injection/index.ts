export * from './createInjectionHooks';
export * from './useInjection';
