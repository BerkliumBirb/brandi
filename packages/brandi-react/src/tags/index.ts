export * from './tagged';
export * from './TagsContext';
export * from './TagsProvider';
export * from './useTags';
