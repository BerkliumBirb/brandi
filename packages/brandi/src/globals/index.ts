export * from './entitiesRegistry';
export * from './injectsRegistry';
export * from './tagsRegistry';
