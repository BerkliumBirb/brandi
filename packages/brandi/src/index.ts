export { Container, createContainer } from './container';
export type {
  Tag,
  Token,
  TokenType,
  TokenTypeMap,
  TokenValue,
  RequiredToken,
  OptionalToken,
} from './pointers';
export { tag, token } from './pointers';
export { injected, tagged } from './registrators';
export type { Factory, ResolutionCondition } from './types';
