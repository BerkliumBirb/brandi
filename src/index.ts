export * from './container';
export * from './registrators';
export * from './token';
export * from './types';
