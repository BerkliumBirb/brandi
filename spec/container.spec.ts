import { Container, token } from '../src';

describe('container', () => {
  it('returns a value from the parent container', () => {
    const someValue = 1;

    const tokens = {
      someValue: token<number>('someValue'),
    };

    const parentContainer = new Container();
    parentContainer.bind(tokens.someValue).toValue(someValue);

    const childContainer = new Container(parentContainer);

    expect(childContainer.get(tokens.someValue)).toBe(someValue);
  });

  it("rebinds a parent container's binding in the child container", () => {
    const someValue = 1;
    const anotherValue = 2;

    const tokens = {
      someValue: token<number>('someValue'),
    };

    const parentContainer = new Container();
    parentContainer.bind(tokens.someValue).toValue(someValue);

    const childContainer = new Container(parentContainer);
    childContainer.bind(tokens.someValue).toValue(anotherValue);

    expect(parentContainer.get(tokens.someValue)).toBe(someValue);
    expect(childContainer.get(tokens.someValue)).toBe(anotherValue);
  });

  it('throws when the token was not bound', () => {
    const tokens = {
      some: token<unknown>('some'),
    };

    const container = new Container();

    expect(() => container.get(tokens.some)).toThrowErrorMatchingSnapshot();
  });

  it("returns an unlinked container from 'copy' method", () => {
    const parentValue = 1;
    const someValue = 2;
    const anotherValue = 3;
    const copiedValue = 4;

    const tokens = {
      parent: token<number>('parent'),
      original: token<number>('original'),
      copied: token<number>('copied'),
    };

    const parentContainer = new Container();
    parentContainer.bind(tokens.parent).toValue(parentValue);

    const originalContainer = new Container(parentContainer);
    originalContainer.bind(tokens.original).toValue(someValue);

    const copiedContainer = originalContainer.copy();
    copiedContainer.bind(tokens.original).toValue(anotherValue);
    copiedContainer.bind(tokens.copied).toValue(copiedValue);

    expect(originalContainer.get(tokens.original)).toBe(someValue);
    expect(() =>
      originalContainer.get(tokens.copied),
    ).toThrowErrorMatchingSnapshot();

    expect(copiedContainer.get(tokens.parent)).toBe(parentValue);
    expect(copiedContainer.get(tokens.original)).toBe(anotherValue);
    expect(copiedContainer.get(tokens.copied)).toBe(copiedValue);
  });

  it('captures the container state to a snapshot', () => {
    const someValue = 1;
    const anotherValue = 2;
    const additionalValue = 3;

    const tokens = {
      some: token<number>('some'),
      additional: token<number>('additional'),
    };

    const container = new Container();
    container.bind(tokens.some).toValue(someValue);

    container.capture();

    container.bind(tokens.some).toValue(anotherValue);
    container.bind(tokens.additional).toValue(additionalValue);

    container.restore();

    expect(container.get(tokens.some)).toBe(someValue);
    expect(() =>
      container.get(tokens.additional),
    ).toThrowErrorMatchingSnapshot();

    container.bind(tokens.some).toValue(anotherValue);
    container.bind(tokens.additional).toValue(additionalValue);

    container.restore();

    expect(container.get(tokens.some)).toBe(someValue);
    expect(() =>
      container.get(tokens.additional),
    ).toThrowErrorMatchingSnapshot();
  });

  it('logs error when trying to restore a non-captured container state', () => {
    const spy = jest.spyOn(console, 'error').mockImplementationOnce(() => null);

    const container = new Container();
    container.restore();

    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy.mock.calls[0]?.[0]).toMatchSnapshot();
  });
});
