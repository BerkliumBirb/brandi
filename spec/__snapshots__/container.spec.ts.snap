// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container captures the container state to a snapshot 1`] = `"No matching bindings found for 'additional' token."`;

exports[`container captures the container state to a snapshot 2`] = `"No matching bindings found for 'additional' token."`;

exports[`container logs an error when trying to restore a non-captured container state in 'development' env 1`] = `"Error: It looks like a trying to restore a non-captured container state. Did you forget to call 'capture()' method?"`;

exports[`container returns an unlinked container from 'clone' method 1`] = `"No matching bindings found for 'copied' token."`;

exports[`container throws when the token was not bound 1`] = `"No matching bindings found for 'some' token."`;
